libsolidity/codegen/ir/IRGeneratorForStatements.cpp

```diff
@@ -234,12 +234,82 @@ void IRGeneratorForStatements::endVisit(Return const& _return)
 
 void IRGeneratorForStatements::endVisit(UnaryOperation const& _unaryOperation)
 {
-	if (type(_unaryOperation).category() == Type::Category::RationalNumber)
+	auto const& opType = type(_unaryOperation);
+	Token const op = _unaryOperation.getOperator();
+
+	if (opType.category() == Type::Category::RationalNumber)
+	{
 		defineExpression(_unaryOperation) <<
-			formatNumber(type(_unaryOperation).literalValue(nullptr)) <<
+			formatNumber(opType.literalValue(nullptr)) <<
 			"\n";
+	}
+	else if (opType.category() == Type::Category::Integer)
+	{
+		string intermediateValue = m_context.newYulVariable();
+
+		if (op == Token::Inc || op == Token::Dec)
+		{
+			auto const* intType = dynamic_cast<IntegerType const*>(&opType);
+
+			solAssert(!!m_currentLValue, "LValue not retrieved.");
+			string lvalue = m_currentLValue->retrieveValue();
+			m_code << "let " << intermediateValue << " := ";
+
+			if (op == Token::Inc)
+				m_code <<
+					m_utils.overflowCheckedUIntAddFunction(intType->numBits());
```

Comment:
It probably makes sense to define increment and decrement functions.

Reply:
really? They seem rather minor to me.. given that it's basically just one function call already.

Reply:
Wait, do you mean in solidity or in c++

Reply:
I mean new yul functions